apple/Sources/FerrostarMapLibreUI/NavigationMapView.swift

```diff
@@ -5,91 +5,81 @@ import MapLibreSwiftDSL
 import MapLibreSwiftUI
 
 public struct NavigationMapView: View {
+    
+    @Environment(\.colorScheme) var colorScheme
+    
     let lightStyleURL: URL
     let darkStyleURL: URL
     // TODO: Configurable camera and user "puck" rotation modes
-
-    @ObservedObject var navigationState: FerrostarObservableState
-
-    @State private var camera: MapView.Camera
-
-    // TODO: Determine this automatically
-    private var useDarkStyle: Bool {
-        return false
+    
+    private var navigationState: FerrostarObservableState?
+    @Binding private var camera: MapViewCamera
+    private var previewRoutes: [Route]?
```

Comment:
It feels like we're now mixing a few different paradigms somewhat dangerously. Here are my concerns:

1. It's not clear where the preview routes will be provided from yet, nor what they will actually do.
2. It is not very clear any more what purpose(s) this view has (this seems half implemented right now so maybe that's why). I'm actually quite OK with it being dual purpose (ex: first displaying route previews and then selecting one and initiating navigation with the selected route), the expected behaviors and responsibilities aren't very clear at the moment, and it still (I think?) is receiving the main navigation state from the outside, even though that state is no longer observable (discussed in another comment). Seems a bit muddled, so let's try to clean this up.
3. I'm not sure what the intended state and update patterns are anymore. This doesn't seem to be purely external (let bindings of value types) nor does it seem to follow the usual mutability pattern with either ref types and observation or `@State` and interior mutability on value types.

Reply:
This is the other big discussion we need to have. How the FerrostarObservableState competes with the MapViewCamera (when a user diverges from the what the navigation is recommending) and any routes being displayed before navigation actually starts.

I feel like in a closed source library, this would all be bundled directly into ferrostar. However, like we discussed, there may be a nice way to let the UI handle itself and let the ferrostarCore publish its "recommendations" based on what it's doing.

Reply:
To better illustrate an idea on this. Think about something like this on the UI. Where the `mapViewCamera` is managed by the UI and its state changes based on what the user wants. If it's set to the navigation specific state, then it just uses the ferrostar camera.

```swift
var mapViewCamera: MapCamera = .default()
var actualCamera: MapCamera {
     if mapViewCamera.state == .userFollowing {
         return ferrostarCore.state.camera
     }
     return mapViewCamera
}
```

(Apologies for the likely very incorrect/loose example, just wanted to get the idea posted for discussion).

Reply:
Yeah, that example makes total sense, and I agree this is really hard to do 😂 We need a way to allow the user to change the viewport AND to snap to something else when the user taps an appropriate button. I've also opened issue #38, which we could use to further discuss how to implement this UI-side.

> How the FerrostarObservableState competes with the MapViewCamera (when a user diverges from the what the navigation is recommending) and any routes being displayed before navigation actually starts.

My original implementation is definitely "wrong" as we should allow for such opinions rather than purely binding.

So I think we're completely agreed at a high level. Let me try to clarify my concerns now, as I don't think I did a good job on the first try.

> 1. It's not clear where the preview routes will be provided from yet, nor what they will actually do.

I still don't get where `previewRoutes` come from, nor why there should be multiple preview routes as part of a navigation map view. I think what you're going for is an Apple Maps-style experience where you see several routes suggested, select one, and seamlessly transition to navigation. I agree with this approach. However, I am not convinced that a `private var` of type `[Route]?` is the best expression of this. It feels like we need internal `@State` which reacts to the user's actions, and is an enumeration that initially contains the alternatives and then transitions to live navigation (or something like that).

I don't think you intended for this PR to get that complex, so perhaps we can just remove this for the moment and revisit later when we actually need this?

> 2. It is not very clear any more what purpose(s) this view has (this seems half implemented right now so maybe that's why)..... <snipped; can re-read the first comment if needed>

My comments on 1 should clarify this. It seems like it's half-implemented so let's punt for this PR and merge the subset that's fully baked.

> 3. I'm not sure what the intended state and update patterns are anymore. This doesn't seem to be purely external (let bindings of value types) nor does it seem to follow the usual mutability pattern with either ref types and observation or @State and interior mutability on value types.

I won't belabor this anymore as the parts relevant to preview routes and camera are discussed above. It's also aimed at the changes to the observable state, and I'll continue discussion around that in the other thread.

Hopefully my questions are a bit clearer now :)

Reply:
Definitely 😆. I would say not even half baked. It's removed for now.

For future discussion, we might also consider simply passing in whatever style layers they want before and during navigation in addition to the required navigation route. I can see this being useful for:

1. A real implementation of "preview routes".
2. Markers and other customizations on the map while navigating or before. 
3. Etc?

Reply:
Yes, agreed. I think we'll get more dynamic on styling as we go :) Thanks for bringing that up!